### ISSUE TITLE
False postive in `UnusedParameter` rule

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#5860]
## Expected Behavior
```kotlin
@ParameterizedTest(
    name = "display msg = {0}",
)
@CsvSource(value = ["'1st test'", "'2nd test"])
fun test(displayMsg: String) {
    assert(true)
}
```
This code should report `UnusedParameter` as we are using `displayMsg` for displaying the name(remember removing the parameter `displayMsg` doesn't replace `{0}` with the passed values from `CsvSource`)

## Observed Behavior
Detekt is reporting `UnusedParameter` for `displayMsg`

## Your Environment
Latest main branch of the detekt

**Comment by cortinico** [comment]
> This is a junit specific thing. I don't think we should care.

Yup this + `allowedNames` can effectively solve this is you agree on a standardized name to use for JUnit test display names

**Comment by atulgpt** [comment]
`allowedNames` sounds reasonable to me. Closing this issue. Feel free to reopen if wants to keep it open for more feedback
